smartsim/control/controller.py

```diff
@@ -378,22 +465,28 @@ def _launch_step(self, job_step, entity):
             logger.debug(f"Launching {entity.name}")
             self._jobs.add_job(job_step.name, job_id, entity)
 
-    def _create_batch_job_step(self, entity_list):
+    def _create_batch_job_step(self, entity):
         """Use launcher to create batch job step
 
-        :param entity_list: EntityList to launch as batch
-        :type entity_list: EntityList
+        :param entity_list: Entity to launch as batch
+        :type entity_list: Entity
         :return: job step instance
         :rtype: Step
         """
         batch_step = self._launcher.create_step(
-            entity_list.name, entity_list.path, entity_list.batch_settings
+            entity.name, entity.path, entity.batch_settings
         )
-        for entity in entity_list.entities:
-            # tells step creation not to look for an allocation
+        if isinstance(entity, EntityList):
```

Comment:
Why is the condition here needed? Is it ever the case that an object is launched as a batch besides an ``EntityList ``?

Reply:
That's the case for Ray, when we launch as batch. Two batch files are created, one for head and one for workers, but both head and workers inherit from `Model`, as each one is only one executable.